src/cli/Utils.cpp

```diff
@@ -108,6 +109,22 @@ namespace Utils
         TextStream out(outputDescriptor);
         TextStream err(errorDescriptor);
 
+        QFileInfo dbFileInfo(databaseFilename);
```

Comment:
Might want to throw in here a check of the canonical file path. That function call returns empty string if the fully resolved file does not exist.

Reply:
Interestingly, with the following patch:

```patch
diff --git a/src/cli/Utils.cpp b/src/cli/Utils.cpp
index b686b155..fede0b29 100644
--- a/src/cli/Utils.cpp
+++ b/src/cli/Utils.cpp
@@ -110,7 +110,9 @@ namespace Utils
         TextStream err(errorDescriptor);
 
         QFileInfo dbFileInfo(databaseFilename);
-        if (!dbFileInfo.exists()) {
+        qInfo("Path: %s", databaseFilename.toStdString().c_str());
+        qInfo("Canonical path: %s", dbFileInfo.canonicalPath().toStdString().c_str());
+        if (dbFileInfo.canonicalPath().isEmpty()) {
             out << QObject::tr("Failed to open database file %1: not found").arg(databaseFilename) << endl;
             return {};
         }
```

I get

```
$ ./src/cli/keepassxc-cli show /tmp/bar foo
Path: /tmp/bar
Canonical path: .
Failed to open database file /tmp/bar: not a plain file
```

So `canonicalPath` returns `.` for a file that does not exist? Doesn't agree with the documentation.

Reply:
You want this function: https://doc.qt.io/qt-5/qfileinfo.html#canonicalFilePath

Reply:
That works, thanks.